blog/_src/posts/2017-09-27-tutorial-contributing-to-racket.md

```diff
@@ -0,0 +1,414 @@
+    Title: Tutorial: Contributing to Racket
+    Date: 2017-09-27T01:07:20
+    Tags: 
+
+*posted by Ben Greenman*
+
+Racket is on GitHub and contributions are always welcome.
+In the spirit of jolly cooperation, this post describes how to submit a quality pull request to a Racket package.
+
+<!-- more -->
+
+What follows is a collection of advice and best-practices for contributing to a
+ Racket package (as of Fall 2017).
+The broad outline for "how to contribute" is the same for Racket as for any
+ mid-sized software project, but there are some important details.
+The real purpose of this post is to list those details in one easy-to-find place.
+
+> See also: [Contributing to Racket](http://blog.racket-lang.org/2012/11/tutorial-contributing-to-racket.html) (the 2012 edition)
+> and [Guide for Infrequent Contributors](http://www.greghendershott.com/2013/04/a-guide-for-infrequent-contributors-to-racket.html)
+
+## Recipe for a Pull Request
+
+- _Step 0_: Get the Racket source code
+- _Step 1_: Fork the code for the package you want to edit
+- _Step 2_: Find the code you need to change
+- _Step 3_: Make changes, write tests, add documentation
+- _Step 4_: Test the changes
+- _Step 5_: Submit the pull request
+
+_Note:_ if something goes wrong following these instructions, ask for
+ help via [email](https://groups.google.com/forum/#!forum/racket-users),
+ [IRC](https://botbot.me/freenode/racket/),
+ and/or [Slack](http://racket-slack.herokuapp.com/).
+
+
+## Step 0. Get the Racket source code
+
+The zero-th step in submitting a pull request is getting a recent copy of Racket.
+The simplest way to be up-to-date is to clone Racket from GitHub and build it from source.
+
+```
+$ git clone https://github.com/racket/racket
+$ cd ./racket
+$ make
+```
+<br/>
+
+Running `make` will take some time.
+Once it finishes, you will have a fresh copy of Racket.
+
+```
+$ ./racket/bin/racket --version
+Welcome to Racket v6.10.0.3.
+```
+<br/>
+
+#### Variable Declarations
+
+To ground the discussion for the rest of this post, we need to define a few
+ variables.
+Let:
+
+- `RKT_DIR/` be the directory created (above) by running `git clone`
+- `my-raco` be the executable at `RKT_DIR/racket/bin/raco`
+- `<PKG>` be the name of the package you want to contribute to
+- `my-branch` be any name you feel like using for a Git branch
+
+
+## 1. Fork the package code
```

Comment:
I wouldn’t expect you to delve into Git basics here, though I recommend having some link to a crash course that the curious / motivated future contributor can follow.